libvips/conversion/flatten.c

```diff
@@ -325,10 +325,13 @@ vips_flatten_build(VipsObject *object)
 	/* Is max-alpha unset? Default to the correct value for this
 	 * interpretation.
 	 */
-	if (!vips_object_argument_isset(object, "max_alpha"))
+	if (!vips_object_argument_isset(object, "max_alpha")) {
 		if (in->Type == VIPS_INTERPRETATION_GREY16 ||
 			in->Type == VIPS_INTERPRETATION_RGB16)
 			flatten->max_alpha = 65535;
+		else if (in->Type == VIPS_INTERPRETATION_scRGB)
+			flatten->max_alpha = 1.0;
+	}
```

Comment:
We have this code in 4 places now. How about adding something like:

```C
double
vips_image_guess_max_alpha(const VipsImage *image)
{
}
```

To `iofuncs/header.c`, just after `vips_image_guess_interpretation()`?

Reply:
Yeah, that's a good idea 👍